Purchases/Purchasing/RCAttributionFetcher.m

```diff
@@ -163,23 +174,38 @@ - (void)postAttributionData:(NSDictionary *)data
         }
     }
 }
+- (void)postAppleSearchAdsAttributionCollectionIfNeeded {
+#if APP_TRACKING_TRANSPARENCY_AVAILABLE
+    if (@available(iOS 14, macos 11, tvos 14, *)) {
+        id<FakeATTrackingManager> trackingManagerClass = (id<FakeATTrackingManager>)NSClassFromString(@"ATTrackingManager");
+
+        if (trackingManagerClass) {
+            NSInteger authorizationStatus = [trackingManagerClass trackingAuthorizationStatus];
+            BOOL authorized = authorizationStatus == FakeATTrackingManagerAuthorizationStatusAuthorized;
+            if (!authorized) {
+                return;
+            }
+        }
+    }
+#endif
 
-- (void)postAppleSearchAdsAttributionCollection {
     NSString *latestNetworkIdAndAdvertisingIdSentToAppleSearchAds = [self
         latestNetworkIdAndAdvertisingIdentifierSentForNetwork:RCAttributionNetworkAppleSearchAds];
-    if (latestNetworkIdAndAdvertisingIdSentToAppleSearchAds == nil) {
-        [self adClientAttributionDetailsWithCompletionBlock:^(NSDictionary<NSString *, NSObject *> *_Nullable attributionDetails,
-                                                              NSError *_Nullable error) {
-            NSArray *values = [attributionDetails allValues];
-
-            bool hasIadAttribution = values.count != 0 && [values[0][@"iad-attribution"] boolValue];
-            if (hasIadAttribution) {
-                [self postAttributionData:attributionDetails
-                              fromNetwork:RCAttributionNetworkAppleSearchAds
-                         forNetworkUserId:nil];
-            }
-        }];
+    if (latestNetworkIdAndAdvertisingIdSentToAppleSearchAds != nil) {
+        return;
     }
+
+    [self adClientAttributionDetailsWithCompletionBlock:^(NSDictionary<NSString *, NSObject *> *_Nullable attributionDetails,
+                                                          NSError *_Nullable error) {
+        NSArray *values = [attributionDetails allValues];
+
+        bool hasIadAttribution = values.count != 0 && [values[0][@"iad-attribution"] boolValue];
+        if (hasIadAttribution) {
+            [self postAttributionData:attributionDetails
+                          fromNetwork:RCAttributionNetworkAppleSearchAds
+                     forNetworkUserId:nil];
+        }
+    }];
 }
```

Comment:
this bit is the same as before, with the only change being that I moved from having everything in an `if` to an early return for readability and consistency with the previous block